// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_outer_classname = "CatDogProto";
option objc_class_prefix = "HLW";

package cat_dog;

// The greeting service definition.
service Animal_classif {
  // Sends a greeting
  rpc cat_or_dog (ImageRequest) returns (AnimalResponse) {}
}

// The request message containing the user's name.
message AnimalResponse {
  string animal = 1;
}

// The response message containing the greetings
message ImageRequest {
    uint32 width = 1;
    uint32 height = 2;
    bytes image_red = 3;
    bytes image_green = 4;
    bytes image_blue = 5;
}

